### PR TITLE
User registration database setup

### PR DESCRIPTION
**Pull request by Aisha-Yusuff**
- Connect backend to database
- Create mongodb helper to connect tests to test database
- Create test for UserSchema
- Create UserSchema in models
- Create a test to save a new user to check connection
- Create beforeEach hook to drop collection before test

### CODE REVIEW

File: server/spec/models/user.spec.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+const mongoose = require("mongoose");
```
Comment (Hafizalimi7) [inline server/spec/models/user.spec.js:1]: good amount of tests !

File: server/models/user.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const mongoose = require("mongoose");
+
```
Comment (Hafizalimi7) [inline server/models/user.js:2]: nice !

File: server/spec/models/user.spec.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  it("can save a new user", (done) => {
+    const user = new User({
+      firstName: "Test",
+      lastName: "Example",
+      email: "test@example.com",
+      password: "password123",
+    });
+
+    user.save((err) => {
```
Comment (robimaliqi) [inline server/spec/models/user.spec.js:48]: great tests

**Review by Hafizalimi7 — APPROVED** [review]

File: server/spec/mongodb_helper.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    })
+    .then(() => {
+      console.log("Connected to MongoDB");
+    })
+    .catch((error) => {
+      console.log("MongoDB connection error", error);
+    });
+  done();
+});
+
```
Comment (robimaliqi) [inline server/spec/mongodb_helper.js:17]: nice function to check if we are connected or not

**Review by robimaliqi — APPROVED** [review]
approved

**Pull request merged**